transports/small-webrtc-transport/src/smallWebRTCTransport.ts

```diff
@@ -473,20 +541,27 @@ export class SmallWebRTCTransport extends Transport {
       logger.debug(`Will create offer for peerId: ${this.pc_id}`);
 
       // Send offer to server
-      const response = await fetch(this._webrtcUrl!, {
-        body: JSON.stringify({
-          sdp: offerSdp.sdp,
-          type: offerSdp.type,
-          pc_id: this.pc_id,
-          restart_pc: recreatePeerConnection,
-        }),
-        headers: {
-          "Content-Type": "application/json",
-        },
-        method: "POST",
-      });
+      const request = cloneDeep(this._webrtcRequest);
+      const requestData: {
+        sdp: string;
+        type: string;
+        pc_id: string | null;
+        restart_pc: boolean;
+        requestData?: any;
+      } = {
+        sdp: offerSdp.sdp,
+        type: offerSdp.type as string,
+        pc_id: this.pc_id,
+        restart_pc: recreatePeerConnection,
+      };
+      if (this._webrtcRequest.requestData) {
+        requestData.requestData = this._webrtcRequest.requestData;
+      }
+      request.requestData = requestData;
```

Comment:
this logic here means that client-provided `requestData` will be nested under the field `requestData` such that the server endpoint will look like:

```
@app.post("/api/offer")
async def offer(request: dict, background_tasks: BackgroundTasks):
    client_data = request.get("requestData")
    if client_data:
        doCustomThing(client_data)
    pc_id = request.get("pc_id")
    ...
 ```
 
 So the question is: Do we like the naming of this field? should it just be `data`? or maybe `customData` or clientData`?

Reply:
Names are always hard. But one thing I like about keeping it as `requestData` is that it matches what we are sending from the client side.

Reply:
Now, looking at the code above, it does feel weird to have `requestData.requestData`. It almost confused me until I realized that it’s actually the custom data from the user that we’re sending inside the request.

So, I am open to be convinced to use a different name. 😅